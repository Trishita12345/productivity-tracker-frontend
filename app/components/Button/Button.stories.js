import React from 'react';
import Button from '.';

export default {
  title: 'form/Button',
  component: Button,
};

const Template = args => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  variant: 'primary',
  children: 'Primary Args',
};

export const Secondary = Template.bind({});
Secondary.args = {
  variant: 'secondary',
  children: 'Secondary Args',
};
