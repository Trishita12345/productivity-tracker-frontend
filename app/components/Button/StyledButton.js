import styled from 'styled-components';
import { COLORS } from '../../utils/constants/colorCodes';

const textColor = {
  primary: COLORS.WHITE,
  secondary: COLORS.DARK_GREY,
};
const bgColor = {
  primary: COLORS.DARK_VIOLET,
  secondary: COLORS.LIGHT_GREY,
};
const bgColorActive = {
  primary: COLORS.BLUE,
  secondary: COLORS.GREY,
};

const StyledButton = styled.button`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 13px 19px;
  color: ${({ variant }) => textColor[variant]};

  background: ${({ variant }) => bgColor[variant]};
  border: ${({ variant }) => bgColor[variant]};
  border-radius: 10px;
  filter: drop-shadow(6px 6px 6px rgba(0, 0, 0, 0.25));

  &:active {
    background: ${({ variant }) => bgColorActive[variant]};
    cursor: pointer;
  }

  &:hover {
    cursor: pointer;
  }
`;

export default StyledButton;
