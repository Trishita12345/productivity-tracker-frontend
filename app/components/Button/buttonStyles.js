import { css } from 'styled-components';

const textColor = {
  primary: '#fffaff',
  secondary: '#87797F',
};
const bgColor = {
  primary: '#0f1322',
  secondary: '#F5F5F5',
};
const bgColorActive = {
  primary: '#4e63b2',
  secondary: '#C6C6C6',
};

const buttonStyles = css`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 13px 19px;
  color: ${({ variant }) => textColor[variant]};

  background: ${({ variant }) => bgColor[variant]};
  border-radius: 10px;

  &:active {
    background: ${({ variant }) => bgColorActive[variant]};
    cursor: pointer;
  }

  &:hover {
    cursor: pointer;
  }
`;

export default buttonStyles;
