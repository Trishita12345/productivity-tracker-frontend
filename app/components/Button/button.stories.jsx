import React from "react";
import Button from ".";

export default {
    title: 'form/Button',
    component: Button,
    args: {
        children: 'Click Me'
    }
}

const Template = args => <Button {...args}/>

export const Primary = Template.bind({})
Primary.args = {
    variant: 'primary'
}