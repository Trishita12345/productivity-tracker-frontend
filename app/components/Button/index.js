/**
 *
 * Button.js
 *
 * A common button, if you pass it a prop "route" it'll render a link to a react-router route
 * otherwise it'll render a link with an onclick
 */

import React, { Children } from 'react';
import PropTypes from 'prop-types';

import StyledButton from './StyledButton';
import Wrapper from './Wrapper';

function Button({ onButtonClick, children, variant = 'secondary' }) {
  return (
    <Wrapper>
      <StyledButton onClick={onButtonClick} variant={variant}>
        {Children.toArray(children)}
      </StyledButton>
    </Wrapper>
  );
}

Button.propTypes = {
  onButtonClick: PropTypes.func,
  variant: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default Button;
