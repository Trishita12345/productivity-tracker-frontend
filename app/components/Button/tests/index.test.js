/**
 * Testing our Button component
 */

import React from 'react';
import { render } from 'react-testing-library';

import Button from '../index';

const handleRoute = () => {};
const href = 'http://mxstbr.com';
const children = <h1>Test</h1>;
const renderComponent = (props = {}) =>
  render(
    <Button href={href} {...props}>
      {children}
    </Button>,
  );

describe('<Button />', () => {
  it('should render a <button> tag to change route if the handleRoute prop is specified', () => {
    const { container } = renderComponent({ handleRoute });
    expect(container.querySelector('button')).toBeDefined();
  });

  it('should not adopt a type attribute when rendering an <a> tag', () => {
    const type = 'text/html';
    const { container } = renderComponent({ href, type });
    expect(container.querySelector(`a[type="${type}"]`)).toBeNull();
  });

  it('should not adopt a type attribute when rendering a button', () => {
    const type = 'submit';
    const { container } = renderComponent({ handleRoute, type });
    expect(container.querySelector('button').getAttribute('type')).toBeNull();
  });
});
