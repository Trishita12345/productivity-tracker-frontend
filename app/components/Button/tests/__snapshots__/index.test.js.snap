// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should renders primary as default 1`] = `ShallowWrapper {}`;

exports[`<Button /> should renders secondary if passed as variant 1`] = `ShallowWrapper {}`;
